### PR TITLE
Move the contents of the "Notes on specific metrics" and "Notes on sp…

### LINKED ISSUE #4446: Include "notes on specific metrics" in the data model so they are exported to the reference manual
The "notes on specific metrics" can be removed from the user manual.

### PR DESCRIPTION
**Pull request by fniessink**
…ecific sources" from the user manual to the reference manual. Closes #4446.

### CODE REVIEW

File: docs/src/create_reference_md.py [hunk 1]
```diff
@@ -142,6 +145,8 @@ def source_section(source: Source, source_key: str, level: int) -> str:
     """Return the source as Markdown section."""
     markdown = markdown_header(source.name, level, index=True)
     markdown += markdown_paragraph(source.description)
+    if documentation := (source.documentation and source.documentation.get("generic")):
```
Comment (wkoot) [inline docs/src/create_reference_md.py]: I think `source.documentation.get("generic") if source.documentation` would be more clear

File: docs/src/create_reference_md.py [hunk 2]
```diff
@@ -142,6 +145,8 @@ def source_section(source: Source, source_key: str, level: int) -> str:
...
+        markdown += markdown_paragraph(documentation)
```
Comment (wkoot) [inline docs/src/create_reference_md.py:151]: How/where is non-generic documentation displayed in the docs?

File: components/shared_data_model/src/shared_data_model/sources/jenkins.py [hunk 3]
```diff
@@ -24,6 +24,15 @@ def jenkins_access_parameters(*args, **kwargs):
     return access_parameters(*args, **kwargs)
 
 
+# Put the Jenkins token documentation in a temporary variable that doesn't trigger a security warning so we can
+# suppress the false positive Bandit warning below.
```
Comment (wkoot) [inline components/shared_data_model/src/shared_data_model/sources/jenkins.py:28]: hahaha, yikes

Comment (fniessink) [reply components/shared_data_model/src/shared_data_model/sources/jenkins.py:28]: Yes, I feel dirty.

File: docs/src/create_reference_md.py (hunk 1 above)
Comment (fniessink) [reply docs/src/create_reference_md.py]: That's not valid Python? I've changed it like this so the walrus assignment doesn't depend on the fact that `a and b` returns `b` if `a` and `b` are both true-ish.

```suggestion
   if source.documentation and (documentation := source.documentation.get("generic")):
```

File: docs/src/create_reference_md.py (hunk 2 above)
Comment (fniessink) [reply docs/src/create_reference_md.py:151]: Added a comment.

**Pull request merged**